Manage AWS Spot interruptions (#586)
seqeralabs/docs: platform-cloud/docs/troubleshooting_and_faqs/nextflow.md

**Invoke Nextflow CLI run arguments during Seqera launch**

> From [Nextflow v22.09.1-edge](https://github.com/nextflow-io/nextflow/releases/tag/v22.09.1-edge), you can specify [Nextflow CLI run arguments](https://www.nextflow.io/docs/latest/cli.html?highlight=dump#run) when invoking a pipeline from Seqera. Set the `NXF_CLI_OPTS` environment variable using a [pre-run script](../launch/advanced.mdx#pre-and-post-run-scripts):

Author: ```suggestion
From [Nextflow v22.09.1-edge](https://github.com/nextflow-io/nextflow/releases/tag/v22.09.1-edge), you can specify [Nextflow CLI run arguments](https://www.nextflow.io/docs/latest/cli.html?highlight=dump#run) when invoking a pipeline from Seqera. Set the `NXF_CLI_OPTS` environment variable using a [pre-run script](../launch/advanced#pre-and-post-run-scripts):
```

Nextflow resolves relative paths against the current working directory. In a classic grid HPC, this normally corresponds to a subdirectory of the `$HOME` directory. In a cloud execution environment, however, the path will be resolved relative to the _container file system_, meaning files will be lost when the container is terminated. See [here](https://github.com/nextflow-io/nextflow/issues/2661#issuecomment-1047259845) for more details.

> Specify the absolute path to your persistent storage using the `NXF_FILE_ROOT` environment variable in your [`nextflow.config`](../launch/advanced.mdx#nextflow-config-file) file. This resolves the relative paths defined in your Netflow script so that output files are written to your stateful storage, rather than ephemeral container storage.

Author: ```suggestion
Specify the absolute path to your persistent storage using the `NXF_FILE_ROOT` environment variable in your [`nextflow.config`](../launch/advanced#nextflow-config-file) file. This resolves the relative paths defined in your Netflow script so that output files are written to your stateful storage, rather than ephemeral container storage.
```

- Head Job CPUs: 16
- Head Job Memory: 60000

> - [Pre-run script](../launch/advanced.mdx#pre-and-post-run-scripts): `export NXF_OPTS="-Xms20G -Xmx40G"`

Author: ```suggestion
- [Pre-run script](../launch/advanced#pre-and-post-run-scripts): `export NXF_OPTS="-Xms20G -Xmx40G"`
```

This change occurs because Seqera superimposes its 10 MB default value rather than the value specified in your `nextflow.config`.

> To force the Seqera-invoked job to use your `nextflow.config` value, add the configuration setting in the workspace Launch screen's [**Nextflow config file** field](../launch/launchpad.mdx). For the example above, you would add `aws.client.uploadChunkSize = 209715200 // 200 MB`.

Author: ```suggestion
To force the Seqera-invoked job to use your `nextflow.config` value, add the configuration setting in the workspace Launch screen's [**Nextflow config file** field](../launch/launchpad). For the example above, you would add `aws.client.uploadChunkSize = 209715200 // 200 MB`.
```

**Fusion v1 execution: _Missing output file(s) [X] expected by process [Y]_ error**

> Fusion v1 has a limitation which causes tasks that run for less than 60 seconds to fail as the output file generated by the task is not yet detected by Nextflow. This is a limitation inherited from a Goofys driver used by the Fusion v1 implementation. [Fusion v2](../supported_software/fusion/fusion.mdx) resolves this issue.

Author: ```suggestion
Fusion v1 has a limitation which causes tasks that run for less than 60 seconds to fail as the output file generated by the task is not yet detected by Nextflow. This is a limitation inherited from a Goofys driver used by the Fusion v1 implementation. [Fusion v2](../supported_software/fusion/fusion) resolves this issue.
```

Your Seqera installation knows the nf-launcher image version it needs and specifies this value automatically when launching a pipeline.

> If you're restricted from using public container registries, see Seqera Enterprise release [instructions](https://docs.seqera.io/platform-enterprise/latest/enterprise/release_notes/23.2.mdx#nextflow-launcher-image) for the specific image to set as the default when invoking pipelines.

Author: ```suggestion
If you're restricted from using public container registries, see Seqera Enterprise release [instructions](https://docs.seqera.io/changelog/seqera-enterprise/v25.1) for the specific image to set as the default when invoking pipelines.
```

**Specify Nextflow version**

Each Seqera Platform release uses a specific nf-launcher image by default. This image is loaded with a specific Nextflow version that any workflow run in the container uses by default. Force your jobs to use a newer/older version of Nextflow with one of the following strategies:

> - Use a [pre-run script](../launch/advanced.mdx#pre-and-post-run-scripts) to set the desired Nextflow version. For example: `export NXF_VER=22.08.0-edge`

Author: ```suggestion
- Use a [pre-run script](../launch/advanced#pre-and-post-run-scripts) to set the desired Nextflow version. For example: `export NXF_VER=22.08.0-edge`
```

> - For jobs executing in an AWS Batch compute environment, create a [custom job definition](https://docs.seqera.io/platform-enterprise/latest/enterprise/advanced-topics/custom-launch-container.mdx) which references a different nf-launcher image.

Author: ```suggestion
- For jobs executing in an AWS Batch compute environment, create a [custom job definition](https://docs.seqera.io/platform-enterprise/latest/enterprise/advanced-topics/custom-launch-container) which references a different nf-launcher image.
```
